scalafmt-sbt/src/main/scala/org/scalafmt/sbt/ScalafmtPlugin.scala

```diff
@@ -1,25 +1,169 @@
 package org.scalafmt.sbt
-import scala.meta.internal.tokenizers.PlatformTokenizerCache
-import sbt._
-import Keys._
+
+import org.scalafmt.{Formatted, Scalafmt}
+import org.scalafmt.config.{Config, ScalafmtConfig, ScalafmtRunner}
+import sbt.Keys._
+import sbt.{Def, _}
+import complete.DefaultParsers._
+import sbt.util.Logger
+
+import scala.util.{Failure, Success, Try}
 
 object ScalafmtPlugin extends AutoPlugin {
   override def trigger: PluginTrigger = allRequirements
+
   object autoImport {
-    val scalafmt: Command =
-      Command.args("scalafmt", "run the scalafmt command line interface.") {
-        case (state, args) =>
-          org.scalafmt.cli.Cli.main("--non-interactive" +: args.toArray)
-          PlatformTokenizerCache.megaCache.clear()
-          state
+    lazy val scalafmt = TaskKey[Unit]("scalafmt", "Format Scala sources")
+    lazy val scalafmtCheck = TaskKey[Boolean](
+      "scalafmt-check",
+      "Check that Scala sources is formatted properly")
+    lazy val scalafmtOnCompile =
+      SettingKey[Boolean]("scalafmt-on-compile", "Format source when compiling")
+    lazy val scalafmtConfig =
+      TaskKey[File]("scalafmt-config", "Scalafmtter config file")
+    lazy val scalafmtSbt = TaskKey[Unit]("scalafmt-sbt", "Format SBT sources")
```

Comment:
The `-` separated names is a relic of sbt 0.12, not to be used in sbt 1 anymore. This should be named `scalafmtSbt`, just like the `val`. In fact, you should probably use `taskKey[Unit]("Format sbt sources")` instead. Same for the other keys (e.g., `scalafmtCheck`).